### PR TITLE
Feature/login

### PR DESCRIPTION
**Pull request by rado196**
1) User login functionality.
2) Throw exception instead of sending error with 200 status code.
3) Added middleware that verifies `One of phone and/or email must be provided`.
4) Generate an access token for the user (tokens not yet stored in DB).

### CODE REVIEW

File: apis/core-api/src/services/AuthService.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    try {
+      this.decode(token);
+      return true;
+    } catch (e) {
+      return false;
+    }
+  }
+
+  static async decode<T = any>(token: string): ServiceResultType<T> {
+    return jwt.verify(token, this.getSecret()) as T;
```
Comment (rubenaprikyan) [inline apis/core-api/src/services/AuthService.ts:38]: Why async ?

**Review by rubenaprikyan — COMMENTED** [review]
Please let's move `apis/core-api/services` into `apis/core-api/core/`

And please describe how we are going to authenticate the user ?
It seems we have a single authentication token for the user, how many devices it can be useful?

**Comment by rado196** [comment]
> Please let's move `apis/core-api/services` into `apis/core-api/core/`
> 
> And please describe how we are going to authenticate the user ? It seems we have a single authentication token for the user, how many devices it can be useful?

I’m not storing token in DB for now, but we can store multiple tokens for user.
For now I generating lifetime token, but I think we need to use access_token and refresh_token  to make app more secure.

**Comment by rubenaprikyan** [comment]
> > Please let's move `apis/core-api/services` into `apis/core-api/core/`
> > And please describe how we are going to authenticate the user ? It seems we have a single authentication token for the user, how many devices it can be useful?
> 
> I’m not storing tokens in DB for now, but we can store multiple tokens for user. For now, I generating a lifetime token, but I think we need to use `access_token` and `refresh_token` to make app more secure.

Make sense, let's discuss this later.

File: apis/core-api/src/services/AuthService.ts (hunk 1 above)
Comment (rado196) [reply apis/core-api/src/services/AuthService.ts:38]: I'm already fixed this part :) methods which not use async functions inside are now called synchronously.

**Review by rubenaprikyan — APPROVED** [review]
lgtm!

**Pull request merged**